vendor/views/mixin.py

```diff
@@ -37,23 +59,23 @@ def user_has_product(self):
 
         TODO: move this to some kind of caching or session variable to avoid hitting the DB on every request.
         """
-        site = Site.objects.get_current()
         if self.request.user.is_anonymous:
             self.product_owned = False
         else:
             products = self.get_product_queryset()
-            self.product_owned = self.request.user.customer_profile.filter(site=site).get().has_product(products)
+            self.product_owned = self.request.user.customer_profile.filter(
+                site=self.request.site).get().has_product(products)
```

Comment:
use the new function here to get the site object instead of modifying the request object.

Reply:
changed to: `self.fields['products'].queryset = Product.objects.filter(site=get_site_from_request(request), available=True)`